tests/query/highlights/haskell/test.hs

```diff
@@ -150,12 +152,22 @@ quasiQuotedString = [qq|Some string|]
                    -- ^ @function.call
                       -- ^ @string
 
-higherOrderFn f x = f x
-           -- ^ @function
-             -- ^ @variable
-
 composition f g = f . g
-         -- ^ @function
-           -- ^ @function
                -- ^ @function
                    -- ^ @function
+qualifiedComposition = SomeModule.f . SomeModule.g
+                        -- ^ @namespace
+                               -- ^ @function
+                                      -- ^ @namespace
+                                              -- ^ @function
```

Comment:
I don't think it needs to be tested this carefully. The test right above seems to me, to have covered for this already. Don't need to overly test for `namespace` as well

Reply:
Oh, I thought I had replied to this...
The above test doesn't match `(qualified_variable (variable))`.
But the `@namespace` tests are overkill here, I agree. I've removed them.